correct6x/corrections.py

```
@@ -36,21 +36,41 @@ def compute_ils_correction(image_df):
     def _rolling_avg(df):
         return df.astype(float).rolling(ROLLING_AVG_TIMESPAN, closed='both').mean()
 
+    try:
+        def _get_ILS(row):
+            # if metadata has independent ils band data, use that
+            if row.independent_ils:
+                return imgparse.get_ils(row.image_path)
+            # otherwise, refer to ils clear band data
+            else:
+                ils_str = 'ILS:Clear="'
+                loc = row.XMP.find(ils_str)
+                if loc == -1:
+                    raise imgparse.xmp.XMPTagNotFoundError("Couldn't parse ILS:Clear string from the image file.")
+                start = loc + len(ils_str)
+                end = row.XMP.find('"', start)
+
+                return float(row.XMP[start:end])
+
+        image_df['ILS'] = image_df.apply(_get_ILS, axis=1)
+    except:
+        logger.warning("Skipping ILS normalization")
+        image_df['ILS_ratio'] = 1
+        return image_df, True
```

I'm not sure what the best way to handle missing ILS values would be. To me, if the user indicated that they wanted ILS correction (by not specifying `--no_ils_correct`), missing ILS values should throw a hard error instead of a warning. It's likely that they either don't have an ILS and didn't know they should set that flag (in which case the error will tell them) or something is wrong with their data (in which case they probably wouldn't want the corrections to proceed without ILS). I'm definitely open to other opinions though.

---

This design decision was made based on a conversation with @ZachMarston. Zach, what do you think?

---

Change reverted